Mathlib/Geometry/Manifold/MFDeriv/SpecificFunctions.lean

```diff
@@ -660,6 +660,76 @@ theorem mfderiv_prod_eq_add_apply {f : M × M' → M''} {p : M × M'} {v : Tange
 
 end Prod
 
+section disjointUnion
+
+variable {M' : Type*} [TopologicalSpace M'] [ChartedSpace H M'] {n : WithTop ℕ∞}
+  {E' : Type*} [NormedAddCommGroup E'] [NormedSpace 𝕜 E'] {H' : Type*} [TopologicalSpace H']
+  {J : Type*} {J : ModelWithCorners 𝕜 E' H'}
+  {N N' : Type*} [TopologicalSpace N] [TopologicalSpace N'] [ChartedSpace H' N] [ChartedSpace H' N']
+
+open Topology
+
+variable {s : Set (M ⊕ M')} {p : M ⊕ M'}
```

Comment:
There seems to be a lot of cruft here:

- `J` is shadowed
- No need for smooth structure
- `Topology` is already open
- `N` and `N'` not used

Really a linter should take care of all this for us, but I think the following is sufficient:
```suggestion
variable {M' : Type*} [TopologicalSpace M'] [ChartedSpace H M'] {s : Set (M ⊕ M')} {p : M ⊕ M'}
```